sdk/python/kfp/compiler/compiler.py

```diff
@@ -818,46 +818,64 @@ def _compile(self, pipeline_func):
     """Compile the given pipeline function into workflow."""
     return self.create_workflow(pipeline_func=pipeline_func)
 
-  def compile(self, pipeline_func, package_path, type_check=True):
-    """Compile the given pipeline function into workflow yaml.
+  @staticmethod
+  def write_workflow(workflow: Dict[Text, Any], package_path: Text = None):
```

Comment:
I'm not sure this method should be public.

Reply:
I think here we have two options:
1. We can expose this API so that TFX side can use it to dump workflow spec to arbitrary format, or;
2. we can duplicate this part of logic to TFX's code base, which should not be a very serious problem because this is not very KFP-specific function. It's just a yaml dump utility.

WDYT?

Reply:
I think that the `Compiler` user should not see or handle the compiled pipeline code dictionary. If this is the case, then the user never has anything they can pass to the `write_workflow` method. Thus, I think, it should be private.
What are the cases where TFX code might want to run `write_workflow`, but not `compile`? Can they call private function/method?

Reply:
> I think that the `Compiler` user should not see or handle the compiled pipeline code dictionary. If this is the case, then the user never has anything they can pass to the `write_workflow` method. Thus, I think, it should be private.
> What are the cases where TFX code might want to run `write_workflow`, but not `compile`? Can they call private function/method?

See [here](https://github.com/tensorflow/tfx/blob/master/tfx/orchestration/kubeflow/kubeflow_dag_runner.py#L237). Currently TFX is calling create_workflow in order to plug in PipelineParam dynamically. I think they can call private function if bypass the lint check.

Reply:
> Currently TFX is calling create_workflow

Maybe `create_workflow` should write the pipeline package the same way the `compile` does?

Reply:
That's a good suggestion. Will let you know once I'm done. Thanks!

Reply:
BTW, I'm in favor of extracting the `_write_workflow` function. I just think that it should be private.